### PR TITLE
[BEAM-6493] : Add cookbook and snippets examples in Kotlin

### PR DESCRIPTION
**Pull request by harshithdwivedi**
This is a continuation of #8291 and the PR adds more sample files in Kotlin.
cc : @pabloem

------------------------

Thank you for your contribution! Follow this checklist to help us incorporate your contribution quickly and easily:

 - [x] [**Choose reviewer(s)**](https://beam.apache.org/contribute/#make-your-change) and mention them in a comment (`R: @username`).
 - [x] Format the pull request title like `[BEAM-XXX] Fixes bug in ApproximateQuantiles`, where you replace `BEAM-XXX` with the appropriate JIRA issue, if applicable. This will automatically link the pull request to the issue.
 - [x] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

Post-Commit Tests Status (on master branch)
------------------------------------------------------------------------------------------------

Lang | SDK | Apex | Dataflow | Flink | Gearpump | Samza | Spark
--- | --- | --- | --- | --- | --- | --- | ---
Go | [Build Status](https://builds.apache.org/job/beam_PostCommit_Go/lastCompletedBuild/) | --- | --- | --- | --- | --- | ---
Java | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Apex/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Dataflow/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Flink/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Java_PVR_Flink_Batch/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Java_PVR_Flink_Streaming/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Gearpump/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Samza/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Spark/lastCompletedBuild/)
Python | [Build Status](https://builds.apache.org/job/beam_PostCommit_Python_Verify/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Python3_Verify/lastCompletedBuild/) | --- | [Build Status](https://builds.apache.org/job/beam_PostCommit_Py_VR_Dataflow/lastCompletedBuild/) <br> [Build Status](https://builds.apache.org/job/beam_PostCommit_Py_ValCont/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PreCommit_Python_PVR_Flink_Cron/lastCompletedBuild/) | --- | --- | ---

Pre-Commit Tests Status (on master branch)
------------------------------------------------------------------------------------------------

--- |Java | Python | Go | Website
--- | --- | --- | --- | ---
Non-portable | [Build Status](https://builds.apache.org/job/beam_PreCommit_Java_Cron/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PreCommit_Python_Cron/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PreCommit_Go_Cron/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PreCommit_Website_Cron/lastCompletedBuild/) 
Portable | --- | [Build Status](https://builds.apache.org/job/beam_PreCommit_Portable_Python_Cron/lastCompletedBuild/) | --- | ---

See [.test-infra/jenkins/README](https://github.com/apache/beam/blob/master/.test-infra/jenkins/README.md) for trigger phrase, status and link of all Jenkins jobs.

### CODE REVIEW

**Comment by pabloem** [comment]
very cool Harshit! I'll take a look soon...

**Comment by harshithdwivedi** [comment]
Sounds good, thanks :)

**Comment by harshithdwivedi** [comment]
Hi @pabloem let me know if you have any comments on this

File: examples/kotlin/src/main/java/org/apache/beam/examples/kotlin/cookbook/README.md [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+-->
+
+# "Cookbook" Examples
+
+This directory holds simple "cookbook" examples, which show how to define
+commonly-used data analysis patterns that you would likely incorporate into a
+larger Apache Beam pipeline. They include:
+
+ <ul>
+  <li><a href="https://github.com/apache/beam/blob/master/examples/java/src/main/java/org/apache/beam/examples/cookbook/BigQueryTornadoes.java">BigQueryTornadoes</a>
```
Comment (pabloem) [inline examples/kotlin/src/main/java/org/apache/beam/examples/kotlin/cookbook/README.md]: The links here point to the Java files. Maybe point to the kotlin files?

**Comment by harshithdwivedi** [comment]
Hi Pablo, sorry for the delayed reply.
I've rested the code and it works fine on my end!

File: examples/kotlin/src/main/java/org/apache/beam/examples/kotlin/cookbook/README.md (hunk 1 above)
Comment (harshithdwivedi) [reply examples/kotlin/src/main/java/org/apache/beam/examples/kotlin/cookbook/README.md]: Yeah, I was planning on sending another PR that fixes the Readme issues; there are some more in other readme files as well.
Let me know if that sounds good?

Comment (pabloem) [reply examples/kotlin/src/main/java/org/apache/beam/examples/kotlin/cookbook/README.md]: That sounds good to me.

**Comment by pabloem** [comment]
This is an issue for a later change, but probably the kotlin examples precommit should run as part of the Java precommit. Thoughts?
Anyway - this change lgtm. ShouldI merge it?

**Comment by harshithdwivedi** [comment]
Yeah, I agree.
Let me look into making this work in a new PR.
For now you can go ahead and merge this one

**Pull request merged**